### ISSUE TITLE
Integration tests are flaky

### ISSUE THREAD

**Issue by viccuad** [kubewarden/kubewarden-controller#628]
It seems that the latest runs of the integration tests in main have randomly failed. See:
https://github.com/kubewarden/kubewarden-controller/actions/runs/7639992405
https://github.com/kubewarden/kubewarden-controller/actions/runs/7727101220

The same code does pass the tests and fail. See:
- PR: https://github.com/kubewarden/kubewarden-controller/actions/runs/7755641962
- Merged PR in main: https://github.com/kubewarden/kubewarden-controller/actions/runs/7755641981/job/21151430022

The tests being flaky seem to be:
`AdmissionPolicy controller when creating a validating AdmissionPolicy when the ValidatingWebhookConfiguration is changed`.

Which come from https://github.com/kubewarden/kubewarden-controller/pull/609/commits/38067affa77b6daa2d1f9ba5719546f9cfd952ee.

Note: PRs are gated by the CI tests and can't be merged unless they are green.

## Acceptance criteria

- Find reason why the tests are failing
- Provide a fix for the failures. It may be that the flakiness reappears in the future, which is not part of this card.

**Comment by viccuad** [comment]
Another flaky test run: https://github.com/kubewarden/kubewarden-controller/actions/runs/7781477305/job/21215983095?pr=633

**Comment by fabriziosestito** [comment]
It seems to be a timeout problem we could increase the timeout of the integration tests to remove failures
